src/models/currency_code.rs

```diff
@@ -0,0 +1,8 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CurrencyCode {
+    PLN,
+    //TODO support other currency codes as needed
```

Comment:
Perhaps while we don't have an exhaustive list, we could also support an `Other(String)` variant.

Reply:
will add this soon